api.hpp

```diff
@@ -726,6 +728,17 @@ class Client {
   // Reference to overridable system dependencies
   std::unique_ptr<Sys> sys{new Sys{}};
 
+ protected:
+  struct summary_data {
```

Comment:
Please use CamelCase for class names. Please add a line comment above the struct. I would be happier if you could also comment before each field and say what is the measurement unit. If there is a place where this is already specified, pointing to such place would be good as well.